src/Filter/CountFilter.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Filter;
+
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface as BaseProxyQueryInterface;
+use Sonata\AdminBundle\Form\Type\Filter\NumberType;
+use Sonata\AdminBundle\Form\Type\Operator\NumberOperatorType;
+use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQueryInterface;
+use Symfony\Component\Form\Extension\Core\Type\NumberType as FormNumberType;
+
+final class CountFilter extends Filter
+{
+    public const CHOICES = [
+        NumberOperatorType::TYPE_EQUAL => '=',
+        NumberOperatorType::TYPE_GREATER_EQUAL => '>=',
+        NumberOperatorType::TYPE_GREATER_THAN => '>',
+        NumberOperatorType::TYPE_LESS_EQUAL => '<=',
+        NumberOperatorType::TYPE_LESS_THAN => '<',
+    ];
+
+    public function filter(BaseProxyQueryInterface $query, $alias, $field, $data)
+    {
+        /* NEXT_MAJOR: Remove this deprecation and update the typehint */
+        if (!$query instanceof ProxyQueryInterface) {
+            @trigger_error(sprintf(
+                'Passing %s as argument 1 to %s() is deprecated since sonata-project/doctrine-orm-admin-bundle 3.27'
+                .' and will throw a \TypeError error in version 4.0. You MUST pass an instance of %s instead.',
+                \get_class($query),
+                __METHOD__,
+                ProxyQueryInterface::class
+            ));
+        }
+
+        if (!$data || !\is_array($data) || !\array_key_exists('value', $data) || !is_numeric($data['value'])) {
+            return;
+        }
+
+        $type = $data['type'] ?? NumberOperatorType::TYPE_EQUAL;
+        $operator = $this->getOperator((int) $type);
+
+        // c.name > '1' => c.name OPERATOR :FIELDNAME
+        $parameterName = $this->getNewParameterName($query);
+        $rootAlias = current($query->getQueryBuilder()->getRootAliases());
+        $query->getQueryBuilder()->addGroupBy($rootAlias);
+        $this->applyWhere($query, sprintf('COUNT(%s.%s) %s :%s', $alias, $field, $operator, $parameterName));
+        $query->getQueryBuilder()->setParameter($parameterName, $data['value']);
+    }
+
+    public function getDefaultOptions()
+    {
+        return [
+            'field_type' => FormNumberType::class,
+        ];
+    }
+
+    public function getRenderSettings()
+    {
+        return [NumberType::class, [
+            'field_type' => $this->getFieldType(),
+            'field_options' => $this->getFieldOptions(),
+            'label' => $this->getLabel(),
+        ]];
+    }
+
+    private function getOperator(int $type): string
+    {
+        return self::CHOICES[$type] ?? self::CHOICES[NumberOperatorType::TYPE_EQUAL];
```

Comment:
IMO, instead of silently fall to `NumberOperatorType::TYPE_EQUAL` here, we should throw a `\OutOfBoundsException` exception here if the given type isn't registered.

Reply:
It's similar https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Filter/AbstractDateFilter.php#L209 or https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Filter/NumberFilter.php

Maybe it's better to merge it like this (to keep consistency) and then open a new PR in order to add a deprecation and throw an exception in next major. WDYT ?